src/DataValues/DecimalMath.php

```diff
@@ -59,16 +59,46 @@ public function getUseBC() {
 	 */
 	public function product( DecimalValue $a, DecimalValue $b ) {
 		if ( $this->useBC ) {
-			$scale = strlen( $a->getFractionalPart() ) + strlen( $b->getFractionalPart() );
-			$product = bcmul( $a->getValue(), $b->getValue(), $scale );
+			return $this->productWithBC( $a, $b );
+		}
+		return $this->productWithoutBC( $a, $b );
+	}
+
+	/**
+	 * @param DecimalValue $a
+	 * @param DecimalValue $b
+	 *
+	 * @return DecimalValue
+	 */
+	private function productWithBC( DecimalValue $a, DecimalValue $b ) {
+		$scale = strlen( $a->getFractionalPart() ) + strlen( $b->getFractionalPart() );
+		$product = bcmul( $a->getValue(), $b->getValue(), $scale );
 
-			$sign = $product[0] === '-' ? '' : '+';
+		$sign = $product[0] === '-' ? '' : '+';
 
-			// (Potentially) round so that the result fits into a DecimalValue
-			// Note: Product might still be to long if a*b >= 10^126
-			$product = $this->roundDigits( $sign . $product, 126 );
-		} else {
-			$product = $a->getValueFloat() * $b->getValueFloat();
+		// (Potentially) round so that the result fits into a DecimalValue
+		// Note: Product might still be to long if a*b >= 10^126
+		$product = $this->roundDigits( $sign . $product, 126 );
+
+		return new DecimalValue( $product );
+	}
+
+	/**
+	 * @param DecimalValue $a
+	 * @param DecimalValue $b
+	 *
+	 * @return DecimalValue
+	 */
+	private function productWithoutBC( DecimalValue $a, DecimalValue $b ) {
+		$product = $a->getValueFloat() * $b->getValueFloat();
+
+		// Append .0 for consistency, if the result is a whole number,
+		// but $a or $b were specified with decimal places.
+		if (
+			$product === floor( $product ) &&
+			$a->getFractionalPart() . $b->getFractionalPart() !== ''
+		) {
+			$product .= '.0';
```

Comment:
Maybe good to do an explicit cast to string. I got a bit confused by the type changing implicitly here. Also suspect that when using PHP strict types this will cause some warning/error.

Reply:
Good point… changed to an explicit string conversion.